dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+
+from dags.analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+
+
+class DiscordTransformRawData(TransformRawDataBase):
+    def transform(self, raw_data: list, platform_id: str, period: datetime) -> list[dict]:
+        transformed_data = []
+        for data in raw_data:
+            interactions = []
+            if data.get("replied"):
+                interactions.append({
+                    "name": "reply",
+                    "users_engaged_id": [data.get("replied")],
+                    "type": "receiver"
+                })
+
+            if data.get("user_mentions"):
+                interactions.append({
+                    "name": "mention",
+                    "users_engaged_id": data.get("user_mentions"),
+                    "type": "emitter"
+                })
+
+            if data.get("reactions"):
+                for reaction in data.get("reactions"):
+                    parts = reaction.split(", ")
+                    if len(parts) > 2:
+                        interactions.append({
+                            "name": "reaction",
+                            "users_engaged_id": parts[:-1],
+                            "type": "emitter"
+                        })
+
+            action_name = "message"
+            action_type = "emitter" if not data.get("isGeneratedByWebhook", False) else "webhook"
```

Comment:
Please assign the `botActivity = True` in metadata when the field `isGeneratedByWebhook` was True. It doesn't affect the emitter or receiver. In all cases when there's a document in rawinfos we're assigning the action type to be a message and emitter. 


conditions for botActivity = True: [link](https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=Note%3A%20the%20botActivity%20would%20be%20identified%20based%20on%20the%20member%20being%20a%20bot%20or%20the%20isGeneratedByWebhook%20field)